src/Blocks/components/form/assets/form.js

```diff
@@ -1047,26 +1047,47 @@ export class Form {
 	setLocalStorage() {
 		// If storage is not set in the backend bailout.
 		// Backend provides the ability to limit what tags are allowed to store in local storage.
-		if (this.storage === '') {
+		if (this.storageConfig === '') {
 			return;
 		}
 
-		const storage = JSON.parse(this.storage);
+		const storageConfig = JSON.parse(this.storageConfig);
 
-		const storageData = storage?.data;
-		const storageExpiration = storage?.expiration ?? '30';
+		const allowedTags = storageConfig?.allowed;
+		const expiration = storageConfig?.expiration ?? '30';
 
 		// Missing data from backend, bailout.
-		if (!storageData) {
+		if (!allowedTags) {
+			return;
+		}
+
+		// Bailout if nothing is set in the url.
+		if (!window.location.search) {
 			return;
 		}
 
 		// Get storage from backend this is considered new by the page request.
-		let newStorage = JSON.parse(storageData);
-		newStorage = {
-			...newStorage,
-			timestamp: Date.now(),
-		};
+		const newStorage = {};
+
+		// Find url params.
+		const searchParams = new URLSearchParams(window.location.search);
+		for (const [key, value] of searchParams.entries()) {
+			// Bailout if not allowed or empty
+			if (!allowedTags.includes(key) || value === '') {
+				continue;
+			}
+
+			// Add valid tag.
+			newStorage[key] = value;
+		}
+
+		// Bailout if nothing is set from allowed tags or everything is empty.
+		if (Object.keys(newStorage).length === 0) {
+			return;
+		}
+
+		// Add current timestamp to new storage.
+		newStorage.timestamp = Date.now();
 
 		// Store in a new variable for later usage.
 		const newStorageFinal = {...newStorage};
```

Comment:
```suggestion
		// Store in a new variable for later usage.
		const newStorageFinal = {...newStorage};
		newStorage.timestamp = Date.now();
```
Wouldn't this avoid the delete process on `newStorageFinal`?

Reply:
newStorage is used to compare the localStorages and the newStorageFinal is used to have an reference to the original one. This is ok

Reply:
Yes, but if you change the order, you don't have to delete it.
If you are ok with deleting, all good, this was just a suggestion